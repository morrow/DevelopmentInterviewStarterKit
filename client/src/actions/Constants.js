export const RECEIVE_ME = 'RECEIVE_ME';
export const RECEIVE_PEOPLE = 'RECEIVE_PEOPLE';