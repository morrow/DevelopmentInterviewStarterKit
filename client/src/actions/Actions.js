import * as actions from './Constants'

export function receiveMe(me) {
  return {
    type: actions.RECEIVE_ME,
    me
  }
}

export function receivePeople(people) {
  return {
    type: actions.RECEIVE_PEOPLE,
    people
  }
}

export function showEmailCharacterFrequency(){
  return {
    type: actions.SHOW_EMAIL_CHARACTER_FREQUENCY,
  }
}

export function hideEmailCharacterFrequency(){
  return {
    type: actions.HIDE_EMAIL_CHARACTER_FREQUENCY,
  }
}

export function toggleEmailCharacterFrequency(){
  return {
    type: actions.TOGGLE_EMAIL_CHARACTER_FREQUENCY,
  }
}

export function showPossibleDuplicates(){
  return {
    type: actions.SHOW_POSSIBLE_DUPLICATES,
  }
}

export function hidePossibleDuplicates(){
  return {
    type: actions.HIDE_POSSIBLE_DUPLICATES,
  }
}

export function togglePossibleDuplicates(){
  return {
    type: actions.TOGGLE_POSSIBLE_DUPLICATES,
  }
}