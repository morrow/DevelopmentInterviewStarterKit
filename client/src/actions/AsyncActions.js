import * as actions from './Actions'

export const creds = { credentials: 'same-origin' };

export function fetchMe() {
  return dispatch => {
    return fetch('/api/me.json', creds)
      .then(response => response.json())
      .then(me => dispatch(actions.receiveMe(me)));
  }
}

export function fetchPeople() {
  return dispatch => {
    return fetch('/api/people.json', creds)
      .then(response => response.json())
      .then(people => dispatch(actions.receivePeople(people)))
  }
}