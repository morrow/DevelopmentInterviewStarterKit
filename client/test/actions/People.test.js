import thunk from 'redux-thunk'
import configureStore from 'redux-mock-store'
import configureMockStore from 'redux-mock-store'
import fetchMock from 'fetch-mock'

import { receivePeople } from '../../src/actions/Actions'
import { fetchPeople } from '../../src/actions/AsyncActions'
import { RECEIVE_PEOPLE } from '../../src/actions/Constants'

describe('people actions', () => {

  let store

  beforeEach(() => {
    const mockStore = configureStore()
    store = mockStore({
      people:[]
    })
    store.clearActions();
  });

  describe('receivePeople', () => {
    test('Dispatches the receivePeople action and payload', () => {
      store.dispatch(receivePeople());
      expect(store.getActions()).toMatchSnapshot();
    });
  });

});

describe('async people actions', () => {

  let mockStore, mock_person

 beforeEach(()=>{
    const middlewares = [thunk]
    mockStore = configureMockStore(middlewares)
    mock_person = {
      name: 'test',
      title: 'test',
      email:'test@salesforce.com'
    }
  })

  afterEach(()=> {
    fetchMock.reset()
    fetchMock.restore()
  })

  describe('fetchPeople', ()=> {

    it('creates RECEIVE_PEOPLE after fetching people', () => {
      fetchMock.getOnce('/api/people.json', {
        body: [mock_person],
        headers: {
          'content-type': 'application/json'
        }
      })
      const expectedActions = [
        {
          type: RECEIVE_PEOPLE,
          people: [mock_person]
        },
      ]
      const store = mockStore({ people: [] })
      return store.dispatch(fetchPeople()).then(() => {
        expect(store.getActions()).toEqual(expectedActions)
      })
    });


  });

});