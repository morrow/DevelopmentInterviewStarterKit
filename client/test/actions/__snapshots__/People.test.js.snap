// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`people actions receivePeople Dispatches the receivePeople action and payload 1`] = `
Array [
  Object {
    "people": undefined,
    "type": "RECEIVE_PEOPLE",
  },
]
`;
